queries.py

```diff
@@ -0,0 +1,355 @@
+            for alias in aliases["aliases"].keys():
+                if alias not in all_indexes:
+                    all_indexes.append(alias)
+
+        all_indexes.append(f"{self.index_name_prefix}-*")
+
+        logger.info(f"Exposed indices: {all_indexes}")
+        return all_indexes
+
+    def format_match(
+        self, hit: dict, base: str, collection: str, expanded: bool = False
+    ):
+        src = hit["_source"]
+        res = {
+            "article_title": src.get("article_title"),
+            "normalized_article_title": src.get("normalized_article_title"),
+            "publication_date": src.get("publication_date")[:10]
+            if src.get("publication_date")
+            else None,
+            "indexed_date": src.get("indexed_date"),
+            "language": src.get("language"),
+            "full_langauge": src.get("full_language"),
+            "url": src.get("url"),
+            "normalized_url": src.get("normalized_url"),
+            "original_url": src.get("original_url"),
+            "canonical_domain": src.get("canonical_domain"),
+            "id": urls.unique_url_hash(src.get("url")),
+        }
+        if expanded:
+            res["text_content"] = src.get("text_content")
+            res["text_extraction"] = src.get("text_extraction")
+        return res
+
+    def format_day_counts(self, bucket: list):
+        return {item["key_as_string"][:10]: item["doc_count"] for item in bucket}
+
+    def format_counts(self, bucket: list):
+        return {item["key"]: item["doc_count"] for item in bucket}
+
+    def proxy_base_url(self, req: Request):
+        return f'{str(os.getenv("PROXY_BASE", req.base_url)).rstrip("/")}/{req.scope.get("root_path").lstrip("/")}'  # type: ignore [union-attr]
+
+    def search_overview(self, collection: str, q: str, req: Request):
+        """
+        Get overview statistics for a query
+        """
+        res = self.ES.search(index=collection, body=QueryBuilder(q).overview_query())  # type: ignore [call-arg]
+        if not res["hits"]["hits"]:
+            raise HTTPException(status_code=404, detail="No results found!")
+
+        total = res["hits"]["total"]["value"]
+        tldsum = sum(
+            item["doc_count"] for item in res["aggregations"]["tld"]["buckets"]
+        )
+        base = self.proxy_base_url(req)
+        return {
+            "query": q,
+            "total": max(total, tldsum),
+            "topdomains": self.format_counts(res["aggregations"]["domain"]["buckets"]),
+            "toptlds": self.format_counts(res["aggregations"]["tld"]["buckets"]),
+            "toplangs": self.format_counts(res["aggregations"]["lang"]["buckets"]),
+            "dailycounts": self.format_day_counts(
+                res["aggregations"]["daily"]["buckets"]
+            ),
+            "matches": [
+                self.format_match(h, base, collection) for h in res["hits"]["hits"]
+            ],
+        }
+
+    def search_result(
```

Comment:
I agree- the req is passed in due to the feature I mention (here)[https://github.com/mediacloud/news-search-api/issues/84] - it's entierly vestigial but I wanted to leave it in the first pass here since there's a conversation ongoing.

Reply:
right now I've removed the proxy_base_url path- I'll leave this comment open for the evening if there are thoughts on committing to that